### PR TITLE
Error plots, bug fixing, saving and loading trained DGMs

### PR DESCRIPTION
**Pull request by jadgle**
Code used for transfer learning with the potential V. 
- there was a bug with the resampling 
- added two functions to the DGM class for saving and loading network weights, .config files
- the saving function also saves plots for the error in terms of the loss in space + error w.r.t. reference solution
- there was a bug in the residual computation, as when summing two arrays with shape [N,1] and [N,] respectively, we were obtaining a [N,N] array
- added tunable hyperparameters for the loss components (all set to 1 by default, when the network is created)

### CODE REVIEW

File: neural_mfg.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def __init__(self):
         
         self.gamma = var['mfg_params']['gamma']; # discount factor
         self.alpha = var['mfg_params']['alpha']; # conjestion operator
-        if gamma == 0:
+        if self.gamma > 0:
```
Comment (matteobutano) [inline neural_mfg.py:49]: Why did you change this?

**Review by matteobutano — CHANGES_REQUESTED** [review]

File: neural_mfg.py (hunk 1 above)
Comment (jadgle) [reply neural_mfg.py:49]: I was working on the branch before you updated the code with the fixed bug, the result should be the same, no?

**Review by matteobutano — APPROVED** [review]
